### PR TITLE
Pip sync quiet

### PR DESCRIPTION
**Pull request by jkimbo**
Add `-q`/`--quiet` flag to `pip-sync` command.

**N.B.** wasn't sure exactly how to add tests for this change. If someone has a suggestion happy to add some.

##### Contributor checklist

- [x] Provided the tests for the changes
- [x] Added the changes to CHANGELOG.md
- [x] Requested (or received) a review from another contributor

### CODE REVIEW

**Comment by davidovich** [comment]
This makes sense, and is easy to test. Look at the tests/test_cli.py [here](https://github.com/jazzband/pip-tools/blob/master/tests/test_cli.py#L194-L205). You just need to add a test that uses the quiet flag and evaluate if there is any output.

**Comment by jkimbo** [comment]
@davidovich I've added a test for the sync command. Had to jump through a couple of hoops to get the command to run but not clobber my virtualenv! Anyway let me know if you had any better ideas on how to test it.

**Comment by vphilippon** [comment]
LGTM.
Could you rebase this on master please?
Thanks!

File: tests/test_cli.py [hunk 1]
```diff
@@ -205,6 +207,30 @@ def test_run_as_module_sync():
...
+
+        with mock.patch('piptools.sync.check_call') as check_call:
+            out = runner.invoke(sync_cli, ['--quiet'])
+            assert out.output == ''
+            assert out.exit_code == 0
+            # for every call to pip ensure the `-q` flag is set
+            for call in check_call.call_args_list:
+                assert '-q' in call[0][0]
+
+        with mock.patch('piptools.sync.check_call') as check_call:
```
Comment (jkimbo) [inline tests/test_cli.py]: Done

**Comment by jkimbo** [comment]
@vphilippon rebased on master

**Pull request merged**